viewing_party/party.py

```diff
@@ -0,0 +1,264 @@
+    """
+    if movie in watched_by_friends:
+        return True
+    else:
+        return False
+
+
+def in_users_subscriptions(host, subscriptions):
+    """
+    Takes two parameters, host and subscriptions.
+    Determines whether the movie host is a service subscription.
+    Returns a Boolean.
+    """
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+
+def get_friends_movies(user_data):
+    """
+    Takes in one parameter user_data.
+    Determines which movies friends has watched.
+    Returns list of dictionaries, movies.
+    """
+    titles = set()
+    friends = user_data["friends"]
+    movies = []
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in titles:
+                movies.append(movie)
+                titles.add(movie["title"])
+    return movies
+
+
+def get_available_recs(user_data):
+    """
+    Takes in one parameter user_data.
+    Determines movies user has not watched, at least
+    one friend has watche, and is available to stream.
+    Returns list of dictionaries available_recs.
+    """
+    user_watched = user_data["watched"]
+    friends_movies = get_friends_movies(user_data)
+    available_recs = []
+
+    if len(friends_movies) == 0:
+        return []
+
+    user_watched_titles = set()
+    for movie in user_watched:
+        user_watched_titles.add(movie["title"])
+
+    for movie in friends_movies:
+        if movie["title"] not in user_watched_titles:
+            if in_users_subscriptions(movie["host"], user_data["subscriptions"]):
+                available_recs.append(movie)
+
+    return available_recs
+
+
+# Wave 5
+def get_new_rec_by_genre(user_data):
+    """
+    Takes in one parameter, user_data.
+    Determines movies user has not watched, at least one
+    friend has watched, and matches most frequent genre.
+    Returns list of dictionaries new_rec_by_genre.
+    """
+    user_watched = user_data["watched"]
+    friends_movies = get_friends_movies(user_data)
+
+    if len(user_watched) == 0 or len(friends_movies) == 0:
+        return []
+
+    user_watched_set = set()
+    for movie in user_watched:
+        user_watched_set.add(movie["title"])
+
+    new_rec_by_genre = []
+    for movie in friends_movies:
+        if movie["title"] not in user_watched_set:
+            if movie["genre"] == get_most_watched_genre(user_data):
+                new_rec_by_genre.append(movie)
+    return new_rec_by_genre
+
+
+def get_rec_from_favorites(user_data):
+    """
+    Takes one parameter, user_data.
+    Determines movies in user's list of dictionaries 
+    "favorites," and has not been watched by friends.
+    Returns list of dictionaries rec_from_favorites.
+    """
+    watched = user_data["watched"]
+    favorites = user_data["favorites"]
+
+    if len(watched) and len(favorites)== 0:
+        return []
+
+    friends_movies = get_friends_movies(user_data)
+    rec_from_favorites = []
```

Comment:
we could combine these again so that we don't have an if statement and only one return statement:

```suggestion
    rec_from_favorites = []
    friends_movies = get_friends_movies(user_data)
  
```